### PR TITLE
Set the joint position with clamping

### PR DESCRIPTION
**Pull request by ssr-yuki**
This PR adds a new function to set the joint position.
In the procedure of assignment, the function clamps the joint position if it is out of joint limits, so that the joint position is always in the joint limit.

### CODE REVIEW

File: src/joint/joint.rs [hunk 1]
```diff
@@ -125,6 +125,43 @@ where
...
+    /// // Create rotational joint with Y-axis
+    /// let mut rot = k::Joint::<f64>::new("r0", k::JointType::Rotational { axis: na::Vector3::y_axis() });
+    ///
+    /// let limits = k::joint::Range::new(-1.0, 1.0);
+    /// rot.limits = Some(limits);
+    ///
+    /// // Initial position is 0.0
+    /// assert_eq!(rot.joint_position().unwrap(), 0.0);
+    /// rot.set_joint_position_clamped(2.0);
+    // assert_eq!(rot.joint_position().unwrap(), 1.0);
```
Comment (OTL) [inline src/joint/joint.rs:147]: `//` -> `///`

File: src/joint/joint.rs [hunk 2]
```diff
@@ -125,6 +125,43 @@ where
...
+    /// rot.set_joint_position_clamped(-2.0);
+    /// assert_eq!(rot.joint_position().unwrap(), -1.0);
+    /// ```
+    ///
+    pub fn set_joint_position_clamped(&mut self, position: T) {
+        if let JointType::Fixed = self.joint_type {
```
Comment (OTL) [inline src/joint/joint.rs:153]: Please use is_movable(), and remove unnecessary `else` (L155).

```rust
if !self.is_movable() {
  return;
}
```

File: src/joint/joint.rs [hunk 3]
```diff
@@ -125,6 +125,43 @@ where
...
+            return;
+        } else if let Some(ref range) = self.limits {
```
Comment (OTL) [inline src/joint/joint.rs:155]: I prefer below style to reduce duplication of cache clear.

```rust
let clamped_position = if let Some(ref range) = self.limits {
  range.clamp(position)
} else {
  position
};
self.set_joint_position_unchecked(position);
```

**Review by OTL — CHANGES_REQUESTED** [review]

File: src/joint/joint.rs (hunk 1 above)
Comment (ssr-yuki) [reply src/joint/joint.rs:147]: Sorry, I miss it. Thanks.

File: src/joint/joint.rs (hunk 2 above)
Comment (ssr-yuki) [reply src/joint/joint.rs]: Ok. I found that `set_joint_position` function also use such a if-statement:
https://github.com/OTL/k/blob/b373b6e2c841a7c033685805c8699c8998129ad3/src/joint/joint.rs#L107

I've replaced both of them by `is_movable`.

**Review by OTL — APPROVED** [review]

**Pull request merged**